File: server/bcephal-license-manager-service/src/main/java/com/moriset/bcephal/license/domain/BrowserDataPage.java
/**
 * 
 */
package com.moriset.bcephal.license.domain;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Data
public class BrowserDataPage<B> {

	private int pageSize;

	private int pageFirstItem;

	private int pageLastItem;

	private int totalItemCount;

	private int pageCount;

	private int currentPage;

	private List<B> items;

	public BrowserDataPage() {
		items = new ArrayList<B>(0);
	}

}

File: client/bcephal-client-model/Base/BrowserDataPage.cs
using Bcephal.Models.Billing.Invoices;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Text;

namespace Bcephal.Models.Base
{
    public class BrowserDataPage<T>
    {

        public int PageSize { get; set; }

        public int PageFirstItem { get; set; }

        public int PageLastItem { get; set; }

        public int TotalItemCount { get; set; }

        public int PageCount { get; set; }

        public int CurrentPage { get; set; }

        public ObservableCollection<T> Items { get; set; }

        public BrowserDataPage()
        {
            Items = new ObservableCollection<T>();
        }

        public static implicit operator BrowserDataPage<T>(BrowserDataPage<BillingTemplateBrowserData> v)
        {
            throw new NotImplementedException();
        }
    }
}
